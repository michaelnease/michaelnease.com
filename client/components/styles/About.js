import styled from "styled-components";

export const AboutStyled = styled.section`
  height: 80rem;
  position: relative;
  background: url("/static/images/bio/bio.gif") no-repeat center center;
  background-size: cover;
  border-bottom: 10px solid ${props => props.theme.colors.black};
  @media (max-width: 700px) {
    height: 100rem;
  }
`;

export const Text = styled.div`
  width: 60rem;
  p {
    text-align: justify;
    font-weight: 300;
    font-size: 1.7rem;
    margin-bottom: 20px;
  }
  @media (max-width: 700px) {
    padding: 3rem;
  }
`;

export const Rex = styled.img`
  box-shadow: ${props => props.theme.effects.bs};
  border: 1px solid ${props => props.theme.effects.black};
  float: left;
  width: 20rem;
  margin: ${props => props.theme.spacing.margin};
  position: relative;
  top: 10px;
`;
