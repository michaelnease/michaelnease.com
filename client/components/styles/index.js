export * from "./Header";
export * from "./Quotes";
export * from "./About";
export * from "./Technology";
export * from "./Portfolio";
export * from "./Contact";
export * from "./Footer";
export * from "./Send-Message-Button";

export * from "./Texture";
