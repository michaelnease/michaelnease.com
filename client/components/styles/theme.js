export const theme = {
  colors: {
    black: "#141414",
    white: "#fff",
    gray: "#e0e0e0",
    darkGray: "#a9a9a9"
  },
  spacing: {
    margin: "15px",
    padding: "15px"
  },
  effects: {
    bs: "-5px 5px 5px 0px rgba(0, 0, 0, 0.2)"
  },
  bar: {
    height: "60px"
  }
};
