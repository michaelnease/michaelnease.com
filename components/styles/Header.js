import styled from "styled-components";

export const Bar = styled.header`
  position: sticky;
  top: 0;
  z-index: 100;
  height: ${props => props.theme.bar.height};
  padding: 0 5%;
  width: 100%;
  background-color: ${props => props.theme.colors.black};
  color: ${props => props.theme.colors.white};
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  box-shadow: ${props => props.theme.effects.bs};
`;

export const Invader = styled.h1`
  width: 52px;
  height: 39px;
  background: url("../static/images/header/invaders-nav.svg");
  text-indent: -9999px;
  overflow: hidden;
  transition: background-position 0.3s;
  background-position: ${props => {
    switch (props.invader) {
      case "crab":
        return "0 0";
      case "antenna":
        return "0 -40px";
      case "legs":
        return "0 -80px";
      case "octopus":
        return "0 -120px";
      default:
        return "0 0";
    }
  }};
`;

export const Navigation = styled.nav`
  ul {
    display: flex;
    justify-content: space-between;
  }
  li {
    margin: 0 10px;
    text-transform: uppercase;
  }
  a {
    color: white;
  }
`;
