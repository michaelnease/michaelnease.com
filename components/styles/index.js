export * from "./Header";
export * from "./Quotes";
export * from "./Bio";
export * from "./Technology";
export * from "./Portfolio";

export * from "./Texture";
