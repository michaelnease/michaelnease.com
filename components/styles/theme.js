export const theme = {
  colors: {
    black: "#141414",
    white: "#fff"
  },
  spacing: {
    margin: "15px"
  },
  effects: {
    bs: "-5px 5px 5px 0px rgba(0, 0, 0, 0.2)"
  },
  bar: {
    height: "60px"
  }
};
