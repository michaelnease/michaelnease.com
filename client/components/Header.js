import { useState } from "react";
import { Bar, Navigation, Invader, Hamburger } from "./styles";

const Header = () => {
  const [name, setInvaderName] = useState("");
  const [open, setHamburger] = useState(false);

  const handleMouseEnter = e => {
    setInvaderName(e.target.name);
  };

  const handleHamburgerClick = () => {
    setHamburger(!open);
  };

  return (
    <Bar>
      <a href="#quotes" data-testid="quotes">
        <Invader invader={name}>Michael Nease</Invader>
      </a>
      <Navigation open={open}>
        <ul>
          <li>
            <a
              onMouseEnter={handleMouseEnter}
              name="crab"
              href="#about"
              data-testid="about"
            >
              About
            </a>
          </li>
          <li>
            <a
              onMouseEnter={handleMouseEnter}
              name="antenna"
              href="#technology"
              data-testid="technology"
            >
              Technology
            </a>
          </li>
          <li>
            <a
              onMouseEnter={handleMouseEnter}
              name="legs"
              href="#portfolio"
              data-testid="portfolio"
            >
              Portfolio
            </a>
          </li>
          <li>
            <a
              onMouseEnter={handleMouseEnter}
              name="octopus"
              href="#contact"
              data-testid="contact"
            >
              Contact
            </a>
          </li>
        </ul>
      </Navigation>
      <Hamburger onClick={handleHamburgerClick}>
        <div />
        <div />
        <div />
      </Hamburger>
    </Bar>
  );
};

export default Header;
