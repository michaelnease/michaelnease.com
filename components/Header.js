import { useState } from "react";
import { Bar, Navigation, Invader } from "./styles";

const Header = () => {
  const [name, setInvaderName] = useState("");

  const handleMouseEnter = e => {
    setInvaderName(e.target.name);
  };

  return (
    <Bar>
      <Invader invader={name}>Michael Nease</Invader>
      <Navigation>
        <ul>
          <li>
            <a onMouseEnter={handleMouseEnter} name="crab" href="#about">
              About
            </a>
          </li>
          <li>
            <a onMouseEnter={handleMouseEnter} name="antenna" href="#skill">
              Skills
            </a>
          </li>
          <li>
            <a onMouseEnter={handleMouseEnter} name="legs" href="#portfolio">
              Portfolio
            </a>
          </li>
          <li>
            <a onMouseEnter={handleMouseEnter} name="octopus" href="#contact">
              Contact
            </a>
          </li>
        </ul>
      </Navigation>
    </Bar>
  );
};

export default Header;
