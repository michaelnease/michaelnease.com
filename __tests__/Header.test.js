import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import { render } from "react-testing-library";
import "jest-styled-components";
import { ThemeProvider } from "styled-components";

import { theme } from "../components/styles/theme";
import Header from "../components/Header";

configure({ adapter: new Adapter() });

describe("<Header />", () => {
  const wrapper = shallow(<Header />);

  it("Should render 1 <Invader /> element", () => {
    expect(wrapper.find("Invader").length).toBe(1);
  });

  it("Should render an <Invader /> with text: Michael Nease", () => {
    expect(wrapper.find("Invader").text()).toBe("Michael Nease");
  });

  it("Should render 4 <Navigation /> elements", () => {
    expect(wrapper.find("Navigation ul li").length).toBe(4);
  });

  it("Should render a <Navigation /> elements", () => {
    expect(
      wrapper
        .find("Navigation ul")
        .childAt(0)
        .text()
    ).toBe("About");
    expect(
      wrapper
        .find("Navigation ul")
        .childAt(1)
        .text()
    ).toBe("Skills");
    expect(
      wrapper
        .find("Navigation ul")
        .childAt(2)
        .text()
    ).toBe("Portfolio");
    expect(
      wrapper
        .find("Navigation ul")
        .childAt(3)
        .text()
    ).toBe("Contact");
  });

  it("Should match Snapshot", () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <Header />
      </ThemeProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
