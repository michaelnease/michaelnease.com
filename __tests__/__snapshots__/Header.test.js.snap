// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With React Testing Library Snapshot Should match Snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 100;
  height: 60px;
  width: 100%;
  background-color: #141414;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: -5px 5px 5px 0px rgba(0,0,0,0.2);
}

.c1 {
  width: 52px;
  height: 39px;
  background: url("../static/images/header/invaders-nav.svg");
  text-indent: -9999px;
  overflow: hidden;
  -webkit-transition: background-position 0.4s;
  transition: background-position 0.4s;
  background-position: 0 0;
}

.c2 ul {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 li {
  margin: 0 10px;
  text-transform: uppercase;
}

.c2 a {
  color: white;
}

<header
    class="c0"
  >
    <h1
      class="c1"
    >
      Michael Nease
    </h1>
    <nav
      class="c2"
    >
      <ul>
        <li>
          <a
            href="#about"
            name="crab"
          >
            About
          </a>
        </li>
        <li>
          <a
            href="#skill"
            name="antenna"
          >
            Skills
          </a>
        </li>
        <li>
          <a
            href="#portfolio"
            name="legs"
          >
            Portfolio
          </a>
        </li>
        <li>
          <a
            href="#contact"
            name="octopus"
          >
            Contact
          </a>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
